backend/app/environments/router.py

```diff
@@ -1,17 +1,56 @@
-from app.environments.schema import Environment
-from app.environments.service import EnvironmentService
-from fastapi import APIRouter, Depends
+from uuid import UUID
+
+from fastapi import APIRouter, Depends, Path, status
 from sqlalchemy.orm import Session
+
 from app.database.database import get_db_session
+from app.dependencies import only_for_admin
+from app.environments.enums import PlatformTypes
+from app.environments.schema import CreatePlatform, Environment, Platform
+from app.environments.service import EnvironmentService
 
 router = APIRouter(prefix="/envs", tags=["environments"])
 
 
 @router.get("")
 def get_environments(db: Session = Depends(get_db_session)) -> list[Environment]:
-    return EnvironmentService().get_environments(db)
+    return EnvironmentService(db).get_environments()
 
 
 @router.post("")
 def create_environment(environment: Environment, db: Session = Depends(get_db_session)):
-    return EnvironmentService().create_environment(environment, db)
+    EnvironmentService(db).create_environment(environment)
+
+
+@router.put(
+    "/{environment}/platforms",
+    status_code=status.HTTP_201_CREATED,
+    dependencies=[Depends(only_for_admin)],
+)
+def add_or_update_environment_settings(
+    environment: str,
+    platform: CreatePlatform,
+    db: Session = Depends(get_db_session),
+):
+    EnvironmentService(db).add_or_update_settings(environment, platform)
+
+
+@router.delete(
+    "/platforms/{id}",
+    status_code=status.HTTP_204_NO_CONTENT,
+    dependencies=[Depends(only_for_admin)],
+)
+def delete_environment_settings(
+    platform_id: UUID = Path(alias="id"),
+    db: Session = Depends(get_db_session),
+):
+    EnvironmentService(db).delete_settings(platform_id)
+
+
+@router.get("/{environment}/platforms", dependencies=[Depends(only_for_admin)])
+def get_environment_platforms(
```

Comment:
I would maybe separate the `get all` from the `get_platform_by_name` call. Then you can also use the correct single return type in both routes

Reply:
It was agreed on our call. I can always return a list if it is preferable.

Reply:
we agreed that we would specify a path parameter IIRC? Currently it is a body / query parameter and it is not always required.
The frontend is set up with two separate calls for getting everything and getting a single one.
The single call should use a path parameter as discussed. If I said body / query parameter previously then that is my bad